src/error.rs

```diff
@@ -24,6 +27,10 @@ pub enum EvalError<'tcx> {
     Unimplemented(String),
     DerefFunctionPointer,
     ExecuteMemory,
+    ArrayIndexOutOfBounds(Span, u64, u64),
+    Math(Span, ConstMathErr),
+    InvalidBitShiftRhs(PrimVal),
+    Overflow(PrimVal, PrimVal, mir::BinOp, PrimVal),
```

Comment:
`EvalError::Overflow` seems to only be used for checking if `eval_binop` overflowed, so if `eval_binop` just did overflowing math internally and also returned a boolean whether something overflowed, could we remove `EvalError::Overflow`?